testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveConnectorKerberosSmokeTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hive;
+
+import com.google.common.io.Closer;
+import io.trino.tempto.AfterTestWithContext;
+import io.trino.tempto.BeforeTestWithContext;
+import io.trino.tempto.query.QueryResult;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.tests.product.TestGroups.HIVE_KERBEROS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.fail;
+
+public class TestHiveConnectorKerberosSmokeTest
+{
+    private Closer closer;
+    private ExecutorService executor;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        closer = Closer.create();
+        executor = newSingleThreadExecutor(); // single thread is enough, it schedules the query to cancel
+        closer.register(executor::shutdownNow);
+    }
+
+    @AfterTestWithContext
+    public void cleanUp()
+            throws IOException
+    {
+        executor = null;
+        closer.close();
+        closer = null;
+    }
+
+    @Test(groups = {HIVE_KERBEROS, PROFILE_SPECIFIC_TESTS}, timeOut = 120_000L)
+    public void kerberosTicketExpiryTest()
+            throws Exception
+    {
+        // Session properties to reduce memory usage and also make the query run longer
+        onTrino().executeQuery("SET SESSION scale_writers = false");
+        onTrino().executeQuery("SET SESSION task_scale_writers_enabled = false");
```

Comment:
Will this session be reflected across all the usage of `onTrino`

Reply:
When `onTrino` is closed it closes the connection it was using thus resetting session properties.

I'll need to verify whether the session is shared even across test classes.

cc: @findepi if he knows the answer. I also see lots of other tests use SET SESSION directly without resetting or any form of isolation.

Reply:
So do we close the connection once the a test is completed ?

Reply:
Actually yes, anything we get from TestContext (e.g. `testContext().getDependency(QueryExecutor.class, config)`) gets closed after each test. This is done by tempto, see https://github.com/trinodb/tempto/blob/a2f7ef3b914db2aeb3480cd845929b1ab26103ef/tempto-core/src/main/java/io/trino/tempto/internal/context/GuiceTestContext.java#L135

Thanks @kokosing for explaining how this magic works.